crates/database/src/auction.rs

```diff
@@ -24,6 +24,16 @@ LIMIT 1
     sqlx::query_as(QUERY).fetch_optional(ex).await
 }
 
+pub async fn load_most_recent_id(ex: &mut PgConnection) -> Result<Option<AuctionId>, sqlx::Error> {
```

Comment:
There is already `load_most_recent()`. Couldn't we just use that for the new endpoint.
Looks like fetching the most recent id and then fetch the data with that id is unnecessarily complex.

Reply:
I noticed that we have load_most_recent already implemented which reads the entire auction struct + its id.
I think that it is unnecessary to read the entire auction for this use-case where we need the latest auction id only.

Reply:
The `/latest` endpoint is supposed to return the entire document, though. Not just the id.
Otherwise you'd have to query `/latest` just to get the ID and then `/solve_competition:id` right afterwards which seems unnecessarily annoying.

Reply:
If you look at the endpoint (solve_competition/latest) implementation, it returns the entire document (competition).
The function we are discussing now is the one that gets the auction id. Why would we fetch the entire auction document if only the auction id is needed to  fetch the competition ?

Reply:
My point is why would you do:
1. fetch latest id internally
2. fetch document with latest id and return from request handler

instead of:
1. fetch latest document and return from request handler

You are not forced to reuse the existing `fetch_by_id_or_tx_hash()` function. You can simply do
```sql
SELECT * FROM solver_competitions ORDER BY id DESC LIMIT 1;
```
and return that.